htseq2multx.pl

```diff
@@ -0,0 +1,992 @@
+my $gzipout         = 0;
+my $split_all       = 0;
+my $fastq_files     = [];
+my $index_hash      = {};
+my $bchash          = {};
+my $samplehash      = {};
+my $fastq_multx     = 'fastq-multx';
+my $debug           = 0;
+my $check_deflines  = 0;
+my $tmp_files       = [];
+my($bcfile,$bcfilefqmx,$prefix,$suffix,$format,$gzipin,$deflinesep);
+
+main();
+
+sub main
+  {
+    processOptions();
+
+    my $command = getFastqMultxCommand();
+
+    if($debug)
+      {print STDERR ("#$command\n")}
+
+    runFastqMultx($command);
+  }
+
+
+
+
+
+
+##
+## Methods
+##
+
+sub processOptions
+  {
+    ##
+    ## Parse the command line and handle help/version options
+    ##
+
+    my $OptionHash = {
+                      'version:+'       => \$version,
+                      'h|help:+'        => \$help,
+                      'bcfile=s'        => \$bcfile,
+                      'idxread=i{1,2}'  => $idxread,
+                      'mismatches=i'    => \$mismatches,
+                      'barcodes_at_end' => \$barcodes_at_end,
+                      'prefix=s'        => \$prefix,
+                      'suffix=s'        => \$suffix,
+                      'galaxy'          => \$galaxy,
+                      'sanitize!'       => \$sanitize,
+                      'v|verbose:+'     => \$verbose,
+                      'gzipout'         => \$gzipout,
+                      'split_all'       => \$split_all,
+                      'format=s'        => \$format,
+                      'gzipin'          => \$gzipin,
+                      '<>'              => sub {push(@$fastq_files,$ARG[0])},
+                      'fastq-multx=s'   => \$fastq_multx,
+                      'debug:+'         => \$debug,
+                      'check-deflines'  => \$check_deflines
+                     };
+
+    if(!GetOptions(%$OptionHash))
+      {
+        print STDERR ("Unable to parse command line.\n");
+        exit($GENERIC_ERROR);
+      }
+
+    ##
+    ## Handle the version option
+    ##
+
+    #Emulate barcode_splitter
+    if($version == 1)
+      {print("$BCSVERSION\n")}
+    #Provide all version information
+    elsif($version)
+      {
+        my($multx_version,$is_fqmx,$problem) = getMultxVersion($fastq_multx);
+        my $multx_used = ($is_fqmx ? join('.',@$multx_version) : 'unknown');
+        print("htseq2multx, version $VERSION\n",
+              "barcode_splitter, emulated version $BCSVERSION\n",
+              "fastq-multx, minimum required version $FQMXVERSION\n",
+              "$fastq_multx, version used $multx_used\n");
+      }
+    if($version)
+      {exit($SUCCESS)}
+
+    #Handle the help option
+    if($help == 1)
+      {help() && exit($SUCCESS)}
+    elsif($help)
+      {help2() && exit($SUCCESS)}
+
+    ##
+    ## Ensure basically valid values for required options are supplied
+    ##
+
+    if(scalar(@$fastq_files) == 0)
+      {
+        help(1);
+        print STDERR ("barcode_splitter: error: the following arguments are ",
```

Comment:
Should these error messages have `htseq2multx.pl` instead of `barcode_splitter`?

Reply:
*sigh*. Well, my whole intent was to make this act and look like barcode_splitter.  I don't know if John is specifically parsing any STDERR.  Probably not, but I was trying to make the output the same.  That was my guiding principle.  I bent that WRT your usage suggestion. So, if you think htseq2multx should intentionally be distinct/different, then I'm fine going with that and I would change things everywhere.  Do you think that would be better?

Reply:
I mean, if that's the motivation behind the specific feedback on the 2 issues where you've brought this up, then we should establish this as a change to the requirements:

https://github.com/PrincetonUniversity/htseq/pull/88/files#diff-3c65b2d41a08a66bc1dbaf00ebf795a7R5

Where line 5 states that this was the goal.

Reply:
I addressed the latest issues except for this one, for which I am just waiting on clarification.

Reply:
I thought about that. It does make sense to work as much the same as barcode splitter as possible, but I figured since you renamed the executable we weren't going for complete transparency. At any rate, this doesn't seem like a huge issue, I just wasn't sure if you were copy/pasting and forgot to change it, or if you intentionally left this here. I stay keep it as is if you like. What was the other second issue I brought up? (Sorry if I brought this up before and we already discussed).

Reply:
I don't think any parsing of stderr is occurring (to my knowledge).  I did see an email regarding re-directing stderr to stdout, which _possibly_ might be unwanted.  Previous because was to have a barcode_splitter stdout as a download and important meta-information file, and if unknown errors were being written there, I don't know the consequences.  Perhaps I misinterpreted that email, though.

Reply:
I'll keep it for now and create an issue to revisit it. The other issue was the executable name in the usage output, which I changed as per your suggestion.